### PR TITLE
Style Book: Show tabs and make blocks clickable when entering edit mode from the Styles menu

### LINKED ISSUE #51887: Style Book: When navigating from the sidebar, there are no tabs and blocks cannot be clicked
### Description

The Style Book can also be accessed from the Style menu in the sidebar. When I click on the button, no tabs appear and all blocks are displayed, which is the expected behavior. However, when I click on the canvas, the tabs do not appear and the blocks are not clickable. Reopening the Stylebook from the right-side style sidebar does not solve the problem.

stylebook

I expect that when the canvas is clicked, i.e. in edit mode, the category tabs will appear as before and each block can also be selected.

Strangely enough, once I go back to template mode and reopen the Stylebook from the right side style sidebar again, it appears as intended.

### Step-by-step reproduction instructions

- Access the Styles menu from the sidebar of the site editor.
- Click on the Style Book button.
- In the canvas area, all blocks are displayed, with no tabs (I think this is the expected behavior).
- Click on the canvas area.
- There are no tabs and each block is not clickable (I don't think this is the correct behavior).
- Click the Style Book button on the right sidebar twice to reopen the Style Book.
- As usual, there are no tabs and each block is not clickable.
- Return to the root menu (Desigin) of the site editor and click on the canvas area.
- Click the Style Book button on the right sidebar.
- The Style Book has tabs and you can click on each block.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/54422211/6de2a026-4acd-4583-a7e9-198939a32e09

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by t-hamano**
Fixes: #51887

## What?
This PR fixes three problems when entering edit mode with the Style Book open from the Styles menu.

- Canvas resizing is not possible
- Category tabs are not displayed
- Can't click blocks

edit-style-book

## Why?
As I understand it, the Style Book displayed in the canvas area is based on the [`<GlobalStylesStyleBook />`](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/global-styles/ui.js#L243). In this component, the canvas is resizable, tabs are visible, and blocks are clickable (default behavior).

However, the Styles page overrides this with a Style Book in [`<SidebarNavigationScreenGlobalStyles />`](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js#L166) to [prohibit canvas resizing, hide tabs, and make the entire canvas clickable](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js#L250-L257).

On the Styles page, the URL contains `wp_global_styles`. I believe this is causing this issue because the Style Book in the `<SidebarNavigationScreenGlobalStyles />` [has priority](https://github.com/WordPress/gutenberg/blob/9a6958e6a3e69ecda18b8deb4e1b3cec352e30a0/packages/edit-site/src/components/sidebar/index.js#L47-L49) even when entering edit mode from this page.

## How?
I used `getCanvasMode()` to not render the Style Book in the `<SizebarNavigationScreenGlobalStyles />` when in **edit** mode. This results in the Style Book being _reset_ to its closed state when you open the Style Book on the Styles page, enter edit mode, and return to view mode. Personally, I think this is acceptable, as it is the intended behavior.

## Testing Instructions

- Open the Site Editor and access the Styles menu.
- Click the Style Book button.
- The Style Book will appear on the canvas. There should be no category tab and all blocks should be displayed.
- Click the canvas area.
- The Style Book should have category tabs, clickable blocks, and a resizable canvas.
- Press the Open Navigation button to return to the Styles menu.
- The canvas area **should not show the Style Book, but the default template.**

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/76773d49-3f2b-4c4b-8541-a89f151edb8b

### CODE REVIEW

**Review by carolinan — APPROVED** [review]
I followed the testing instructions and both view and edit mode are working as expected.

File: packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js [hunk 1]
```diff
@@ -170,13 +170,15 @@ export default function SidebarNavigationScreenGlobalStyles() {
...
-			'style-book' ===
-			unlock( select( editSiteStore ) ).getEditorCanvasContainerView(),
-		[]
-	);
+	const { isViewMode, isStyleBookOpened } = useSelect( ( select ) => {
+		const { getCanvasMode, getEditorCanvasContainerView } = unlock(
+			select( editSiteStore )
+		);
+		return {
+			isViewMode: getCanvasMode() === 'view',
```
Comment (ramonjd) [inline packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js]: Nit (and totally optional) - the expression below uses Yoda condition. For neatness, should give them the same style? Either one or the other, I don't believe there's a preferred way. 😄

**Review by ramonjd — COMMENTED** [review]
LGTM

- Clicking category tabs works
- The stylebook iframe is resizeable
- Selecting a block opens the corresponding block style pane

Nice work @t-hamano !

I think it needs a rebase though.

**Comment by t-hamano** [comment]
Thanks for the review!

I used Yoda notation and also resolved the conflict. This PR was conflicting with #52117 which automatically disables Distraction Free Mode on Style Book page.

So please make sure that the behavior is maintained in this PR:

- Open the Site Editor
- Enable Distraction Free Mode
- Return to view mode and open the Styles menu
- Click the Style Book button
- The Style Book will appear on the canvas. There should be no category tab and all blocks should be displayed.
- Click the canvas area.
- The Style Book should have category tabs, clickable blocks, and a resizable canvas.
- **Confirm that a snack bar with the text "Distarction free mode turned off" appears.**
- **Distraction Free Mode should be disabled.**
- Press the Open Navigation button to return to the Styles menu.
- The canvas area should not show the Style Book, but the default template.

https://github.com/WordPress/gutenberg/assets/54422211/cb670a90-dd56-40fb-be7b-243aee8a46bd

**Comment by ramonjd** [comment]
Re-tested.

All is well 😄

**Review by ramonjd — APPROVED** [review]
🙇🏻

**Comment by t-hamano** [comment]
Hmm, several E2E tests are failing, but this does not appear to be due to this PR, as it is also occurring on other PRs. I would like to merge this issue once it is resolved.

**Comment by ramonjd** [comment]
> Hmm, several E2E tests are failing, but this does not appear to be due to this PR, as it is also occurring on other PRs. I would like to merge this issue once it is resolved.

Happening on all branches as far as I'm aware 😢 

Related: https://github.com/WordPress/gutenberg/pull/52167#issuecomment-1620577509

**Comment by t-hamano** [comment]
I consider this PR to be a bug fix, so I will give it a backport label.

**Comment by tellthemachines** [comment]
I just cherry-picked this PR to the update/first-pre-beta-4 branch to get it included in the next release: 621aa9557e

**Pull request merged**